src/app/components/MostRead/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useContext } from 'react';
+import { RequestContext } from '#contexts/RequestContext';
+import useToggle from '#hooks/useToggle';
+import { getMostReadEndpoint } from '#app/lib/utilities/getUrlHelpers/getMostReadUrls';
+import { ServiceContext } from '../../contexts/ServiceContext';
+import Canonical from './Canonical';
+import Amp from './Amp';
+import { ColumnLayout, Size, MostReadData } from './types';
+import MostReadSection from './Section';
+import MostReadSectionLabel from './Label';
+import { WHITE } from '../ThemeProvider/palette';
+
+const blockLevelEventTrackingData = {
+  componentName: 'most-read',
+};
+
+const mostReadAmpPageTypes = ['STY', 'CSP', 'article', 'home'];
+
+interface MostReadProps {
+  data?: MostReadData;
+  columnLayout?: ColumnLayout;
+  size?: Size;
+  mobileDivider?: boolean;
+  backgroundColour?: string;
+}
+
+const MostRead = ({
+  data,
+  columnLayout = 'multiColumn',
+  size = 'default',
+  mobileDivider = false,
+  backgroundColour = WHITE,
+}: MostReadProps) => {
+  const { isAmp, pageType, variant } = useContext(RequestContext);
+  const {
+    service,
+    mostRead: { hasMostRead },
+  } = useContext(ServiceContext);
+
+  const { enabled } = useToggle('mostRead');
+
+  const mostReadToggleEnabled = enabled && hasMostRead;
+
+  // Do not render most read when a toggle is disabled
+  if (!mostReadToggleEnabled) {
+    return null;
+  }
+
+  const endpoint = getMostReadEndpoint({
+    service,
+    variant,
+  });
+
+  // We render amp on ONLY STY, CSP and ARTICLE pages using amp-list.
+  const AmpMostRead = () =>
+    mostReadAmpPageTypes.includes(pageType) ? (
+      <MostReadSection>
+        <MostReadSectionLabel
+          mobileDivider={mobileDivider}
+          backgroundColor={backgroundColour}
+        />
+        <Amp
+          endpoint={`${process.env.SIMORGH_MOST_READ_CDN_URL}${endpoint}`}
+          size={size}
+        />
+      </MostReadSection>
+    ) : null;
+
+  // Do not render on Canonical if data is not provided
```

Comment:
Note, this is a change from the existing behaviour of the legacy container, where the container would fetch the data on the client side. We have decided that the new canonical component will not do any client side data fetching, and will just render the data which has been passed to it. 

The BFF already includes the mostRead response for CPS Assets, Articles and HomePages, so it should be fairly straightforward to get this working once we migrate all pages types to the new component

Reply:
What was the reason for client side fetching? Was it to do with content freshness?

Reply:
I believe it was partly to do with content freshness, and also (I am guessing) something to do with how we used to fetch the data - before the BFF, we only fetched the content from ARES via the .json endpoint, but then the MostRead component would be responsible for fetching its own data? Although I could be completely wrong - @andrewscfc or @amoore108 would you happen to know more?

Reply:
Okay, yeah I guess it depends if it was reliant on content freshness, now that we're relying on the BFF for this data has the content freshness been affected? And was that a major factor for the front pages?

Reply:
Regarding content freshness via the BFF, there is currently a ticket to investigate the delay in time to publish content based on updates to TIPO which could affect `MostRead` but probably more pertinent to `Live` pages:

https://jira.dev.bbc.co.uk/browse/WSTEAMA-378?focusedCommentId=5945427&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-5945427